Lesson-2/BankAcc.cs

```diff
@@ -113,6 +113,36 @@ public override string ToString()
         {
             return $"Тип счёта: {this.Type}\nНомер счёта: {this.Numb}\nБаланс: {this.Balance}";
         }
+
+        public static bool operator ==(BankAcc first, BankAcc second)
+        {
+            return first.Balance== second.Balance;
+        }
+
+        public static bool operator !=(BankAcc first, BankAcc second)
+        {
+            return !(first.Balance == second.Balance);
+        }
+
+        public bool Equals(BankAcc compared)
+        {
+            return this.Balance == compared.Balance;
+        }
+
+        public override bool Equals(object? obj)//есть здесь, только потому что в задании было сказано "переопределить", не "перезагрузить". 
+        {
+            if(obj == null)
+                return false;
+            var notObj = obj as BankAcc;
+            if (notObj is null)
+                return false;
+            return this.Balance == notObj.Balance;
```

Comment:
Можно проще

Reply:
```suggestion
            if (obj is not BankAcc notObj)
                return false;
            return this.Balance == notObj.Balance;
```
Называется - "сопоставление с образцом".